src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -3577,8 +3711,7 @@ def normalize_basis_at_p(e, p, B=QuaternionAlgebraElement_abstract.pair):
 
             # Ensures that (B(f0,f0)/2).valuation(p) <= B(f0,f1).valuation(p)
             if B(f0, f1).valuation(p) + 1 < B(f0, f0).valuation(p):
-                f0 += f1
-                f1 = f0
+                f0, f1 = f0 + f1, f0
```

Comment:
This change is makes the code behave in a very subtle way. It would be better to make it explicit.